### DISCUSSION TITLE
Force row de-selection

### DISCUSSION THREAD

**Discussion by proddy** [table-library/react-table-library#121] [Q&A] [answered]
I'm using `useRowSelect` on a Table to toggle row selection, which works fine. Now I'd like to add a button to reset the selection, i.e. unselect the row. It's purely a visual thing. Is there a way to do this? Like if my select looks like:

```ts
const myselect = useRowSelect(
  data,
  {
    onChange: onSelectChange,
  }
);
```

I was hoping setting `myselect.state.id` to `null` would work.

**Comment by rwieruch** [comment, marked as answer]
You have programmatically access to all functions of the feature. Here is a button which deselects all rows:

```
const myselect = useRowSelect(
  data,
  {
    onChange: onSelectChange,
  }
);

return (
  <>
    <button onClick={() => myselect.fns.onRemoveAll()}>Reset</button>
    
    <Table ... >...</Table>
  </>
);
```

Let me know if this helps!

**Reply by proddy** [reply]
works like a charm, thanks!
